### ISSUE TITLE
DHCP Client fails to get IP address and includes 802.1q header

### ISSUE THREAD

**Issue by btashton** [systemd/systemd#9874]
**systemd version the issue has been seen with**
> systemd 239
-PAM -AUDIT -SELINUX -IMA -APPARMOR -SMACK +SYSVINIT +UTMP -LIBCRYPTSETUP +GCRYPT -GNUTLS -ACL -XZ -LZ4 -SECCOMP +BLKID -ELFUTILS +KMOD -IDN2 -IDN -PCRE2 default-hierarchy=hybrid

**Used distribution**
 > buildroot

**Expected behaviour you didn't see**
 > Router responds to DHCP Discovery packet

**Unexpected behaviour you saw**
 > Router never responded to DHCP Discovery packet and the outbound packet included a 802.1q header.

**Steps to reproduce the problem**
Define a network such as this:
```
/etc/systemd/network/wireless.network 
[Match]
Name=wlan0

[Network]
DHCP=yes

[DHCP]
ClientIdentifier=mac
```

Connect to a stock configured router such as the Apple Airport Express (A1084) or WRT54G with stock settings and firmware.

Sniff from another computer on the wireless network:
```
Frame 77884: 324 bytes on wire (2592 bits), 324 bytes captured (2592 bits) on interface 0
Ethernet II, Src: MurataMa_91:d0:7e (fc:db:b3:91:d0:7e), Dst: Broadcast (ff:ff:ff:ff:ff:ff)
    Destination: Broadcast (ff:ff:ff:ff:ff:ff)
    Source: MurataMa_91:d0:7e (fc:db:b3:91:d0:7e)
    Type: 802.1Q Virtual LAN (0x8100)
802.1Q Virtual LAN, PRI: 6, DEI: 0, ID: 0
    110. .... .... .... = Priority: Internetwork Control (6)
    ...0 .... .... .... = DEI: Ineligible
    .... 0000 0000 0000 = ID: 0
    Type: IPv4 (0x0800)
Internet Protocol Version 4, Src: 0.0.0.0, Dst: 255.255.255.255
User Datagram Protocol, Src Port: 68, Dst Port: 67
Bootstrap Protocol (Discover)
    Message type: Boot Request (1)
    Hardware type: Ethernet (0x01)
    Hardware address length: 6
    Hops: 0
    Transaction ID: 0xa49f4993
    Seconds elapsed: 1
    Bootp flags: 0x0000 (Unicast)
    Client IP address: 0.0.0.0
    Your (client) IP address: 0.0.0.0
    Next server IP address: 0.0.0.0
    Relay agent IP address: 0.0.0.0
    Client MAC address: MurataMa_91:d0:7e (fc:db:b3:91:d0:7e)
    Client hardware address padding: 00000000000000000000
    Server host name not given
    Boot file name not given
    Magic cookie: DHCP
    Option: (53) DHCP Message Type (Discover)
        Length: 1
        DHCP: Discover (1)
    Option: (61) Client identifier
        Length: 7
        Hardware type: Ethernet (0x01)
        Client MAC address: MurataMa_91:d0:7e (fc:db:b3:91:d0:7e)
    Option: (55) Parameter Request List
        Length: 8
        Parameter Request List Item: (1) Subnet Mask
        Parameter Request List Item: (3) Router
        Parameter Request List Item: (12) Host Name
        Parameter Request List Item: (15) Domain Name
        Parameter Request List Item: (6) Domain Name Server
        Parameter Request List Item: (33) Static Route
        Parameter Request List Item: (121) Classless Static Route
        Parameter Request List Item: (42) Network Time Protocol Servers
    Option: (57) Maximum DHCP Message Size
        Length: 2
        Maximum DHCP Message Size: 576
    Option: (12) Host Name
        Length: 9
        Host Name: AHOSTNAME
    Option: (255) End
        Option End: 255

0000   ff ff ff ff ff ff fc db b3 91 d0 7e 81 00 c0 00   ÿÿÿÿÿÿüÛ³.Ð~..À.
0010   08 00 45 c0 01 32 00 00 00 00 40 11 78 fc 00 00   ..EÀ.2....@.xü..
0020   00 00 ff ff ff ff 00 44 00 43 01 1e 9a 12 01 01   ..ÿÿÿÿ.D.C......
0030   06 00 a4 9f 49 93 01 00 00 00 00 00 00 00 00 00   ..¤.I...........
0040   00 00 00 00 00 00 00 00 00 00 fc db b3 91 d0 7e   ..........üÛ³.Ð~
0050   00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00   ................
0060   00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00   ................
0070   00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00   ................
0080   00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00   ................
0090   00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00   ................
00a0   00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00   ................
00b0   00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00   ................
00c0   00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00   ................
00d0   00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00   ................
00e0   00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00   ................
00f0   00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00   ................
0100   00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00   ................
0110   00 00 00 00 00 00 00 00 00 00 63 82 53 63 35 01   ..........c.Sc5.
0120   01 3d 07 01 fc db b3 91 d0 7e 37 08 01 03 0c 0f   .=..üÛ³.Ð~7.....
0130   06 21 79 2a 39 02 02 40 0c 09 41 48 4f 53 54 4e   .!y*9..@..AHOSTN
0140   41 4d 45 ff                                       AMEÿ
```

This discovery message will never be acknowledged by the router.

Use dhclient with default settings and the router will respond.
The DHCP Discovery packet does not include the 802.1q header

```
Frame 4081: 342 bytes on wire (2736 bits), 342 bytes captured (2736 bits) on interface 0
Ethernet II, Src: MurataMa_91:d0:7e (fc:db:b3:91:d0:7e), Dst: Broadcast (ff:ff:ff:ff:ff:ff)
    Destination: Broadcast (ff:ff:ff:ff:ff:ff)
    Source: MurataMa_91:d0:7e (fc:db:b3:91:d0:7e)
    Type: IPv4 (0x0800)
Internet Protocol Version 4, Src: 0.0.0.0, Dst: 255.255.255.255
User Datagram Protocol, Src Port: 68, Dst Port: 67
Bootstrap Protocol (Discover)
    Message type: Boot Request (1)
    Hardware type: Ethernet (0x01)
    Hardware address length: 6
    Hops: 0
    Transaction ID: 0x5ff69035
    Seconds elapsed: 0
    Bootp flags: 0x0000 (Unicast)
    Client IP address: 0.0.0.0
    Your (client) IP address: 0.0.0.0
    Next server IP address: 0.0.0.0
    Relay agent IP address: 0.0.0.0
    Client MAC address: MurataMa_91:d0:7e (fc:db:b3:91:d0:7e)
    Client hardware address padding: 00000000000000000000
    Server host name not given
    Boot file name not given
    Magic cookie: DHCP
    Option: (53) DHCP Message Type (Discover)
        Length: 1
        DHCP: Discover (1)
    Option: (55) Parameter Request List
        Length: 9
        Parameter Request List Item: (1) Subnet Mask
        Parameter Request List Item: (28) Broadcast Address
        Parameter Request List Item: (2) Time Offset
        Parameter Request List Item: (3) Router
        Parameter Request List Item: (15) Domain Name
        Parameter Request List Item: (6) Domain Name Server
        Parameter Request List Item: (12) Host Name
        Parameter Request List Item: (44) NetBIOS over TCP/IP Name Server
        Parameter Request List Item: (47) NetBIOS over TCP/IP Scope
    Option: (255) End
        Option End: 255
    Padding: 000000000000000000000000000000000000000000000000...

0000   ff ff ff ff ff ff fc db b3 91 d0 7e 08 00 45 10   ÿÿÿÿÿÿüÛ³.Ð~..E.
0010   01 48 00 00 00 00 80 11 39 96 00 00 00 00 ff ff   .H......9.....ÿÿ
0020   ff ff 00 44 00 43 01 34 3e 7a 01 01 06 00 5f f6   ÿÿ.D.C.4>z...._ö
0030   90 35 00 00 00 00 00 00 00 00 00 00 00 00 00 00   .5..............
0040   00 00 00 00 00 00 fc db b3 91 d0 7e 00 00 00 00   ......üÛ³.Ð~....
0050   00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00   ................
0060   00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00   ................
0070   00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00   ................
0080   00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00   ................
0090   00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00   ................
00a0   00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00   ................
00b0   00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00   ................
00c0   00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00   ................
00d0   00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00   ................
00e0   00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00   ................
00f0   00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00   ................
0100   00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00   ................
0110   00 00 00 00 00 00 63 82 53 63 35 01 01 37 09 01   ......c.Sc5..7..
0120   1c 02 03 0f 06 0c 2c 2f ff 00 00 00 00 00 00 00   ......,/ÿ.......
0130   00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00   ................
0140   00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00   ................
0150   00 00 00 00 00 00                                 ......
```

What is further odd is the request to a modern router via systemd-networkd which work fine on this hardware do no include this mysterious 802.1q header which is why I suspect it to be related to the failure.  I have not been able to track down where this is getting added to the payload and the only reason I have opened it against systemd is because both udhcp from Busybox and dhclient do not have any issues connecting to the routers on this hardware.

**Comment by yuwata** [comment]
Related to #9794?

**Comment by btashton** [comment]
I thought it might be related, but the padding does not seem to be the cause.

**Comment by ssahani** [comment]
Could we know which router is this ? I kind of confused . Can you look at your router logs and confirm these packets (UDP) even reaching to DHCP server ? If yes  please try to paste those logs that what exact reason it's not responding.

**Comment by yuwata** [comment]
Fix is waiting in #21702.
